torch/_export/passes/replace_set_grad_with_hop_pass.py

```diff
@@ -147,11 +147,21 @@ def _sequential_split_and_maybe_inline_subgraphs(
             need_replacing = True
 
     if need_replacing:
+        # sequential_split returns a new graph module that could have different output 
+        # args names. We need to fix the graph signature.
         new_gm = sequential_split(gm, _is_set_grad_enabled_node)
-
+        
         replace_ctx = contextlib.nullcontext()
+        new_signature = None
         if graph_signature is not None:
-            replace_ctx = new_gm._set_replace_hook(graph_signature.get_replace_hook())  # type: ignore[assignment]
+            new_signature = copy.deepcopy(graph_signature)
+            new_gm_out_node =  next(reversed(new_gm.graph.find_nodes(op="output")))
+            assert new_gm_out_node.op == "output" and len(new_gm_out_node.args[0]) == len(new_signature.output_specs)
+            for arg_node, out_spec in zip(new_gm_out_node.args[0], new_signature.output_specs):
+                if out_spec.arg.name != arg_node.name:
+                    out_spec.arg.name = arg_node.name
```

Comment:
I'm surprised _set_replace_hook doesn't already do this?

Reply:
Yeah, but it can only do its job when the graph module being inplace transformed and graph signature has a correct correspondence.

The reason we need to this mannual mapping is that, sequential_split createa a new graph module that has inconsistent output names with the name stored in graph signature.output_specs. As a result, if we don't mannually override the output names, even if we set the new_graph_module's replace hook to be new_signature's replace_hook, the replace_hook function cannot find the original name in the output_specs cauing the replacement fail silently [here](https://github.com/pytorch/pytorch/pull/129146/files#diff-92420f977c3a02b2deadf6752ce4a9ee601c20612a1a13cc365252eb09410edbL516). One possible better way is we should error out when we cannot find match any old names.

Reply:
yeah that makes sense! thanks for the explanation! 
> One better way is we should error out when we cannot find match any old names in the replace hook to avoid this kind of silent incorrectness.
I'm not sure if erroring will cause any issues.. can do in a separate PR I guess :P